pkg/analyzer/lib/file_system/physical_file_system.dart

```diff
@@ -62,7 +77,14 @@ class PhysicalResourceProvider implements ResourceProvider {
   final AbsolutePathContext absolutePathContext =
       new AbsolutePathContext(io.Platform.isWindows);
 
-  PhysicalResourceProvider(FileReadMode fileReadMode) {
+  factory PhysicalResourceProvider(FileReadMode fileReadMode) {
+    return new PhysicalResourceProvider.at(fileReadMode: fileReadMode);
+  }
+
+  PhysicalResourceProvider.at({String folder, FileReadMode fileReadMode})
```

Comment:
ResourceProvider is used not only for reading state / cache, but also for any file.
So, specializing it so much for state is not right.
Please rollback changes to this file.

Reply:
Could you clarify what's not right? PhysicalResourceProvider provides a way to store state, and this change adds a way for clients to get a tiny bit more control over how it's done rather than rely on platform concepts of a "home".
I don't mind making the change, but I'm worried that this would be brittle if state bleeds into a different location.

Reply:
`ResourceProvider` provides a way to access files and folders.

There is also a small addition to access "state location". Making state location a center of `ResourceProvider` by making a special constructor seems like overkill. The location path could be a named parameter of the existing `PhysicalResourceProvider` constructor.

Reply:
Updated to only have a single constructor.